### ISSUE TITLE
Feature: Append to archive

### ISSUE THREAD

**Issue by xeruf** [borgbackup/borg#6073]
## Have you checked borgbackup docs, FAQ, and open Github issues?

yes

## Is this a BUG / ISSUE report or a QUESTION?

ISSUE

## Describe the problem you're observing.

While collecting data for assembling my borg repository I have found folders that belong together.
Is there a way to amend an existing archive?

**Comment by ThomasWaldmann** [comment]
No, archives are not mutable after creation of them is finished.

There is `borg recreate` though which can do some sort of changes (but not add new files), but even that is not mutating an archive, but just reading it, creating a new one and then renaming the new over the old one.

**Comment by jdchristensen** [comment]
But maybe the more important advice is that the original poster should just create a new archive with the files included, and it will be roughly as fast as hypothetically adding those files to an existing archive, since borg does deduplication.  They can then either delete the original archive, or keep it around until it gets pruned (since it will hardly use any additional space).

**Comment by xeruf** [comment]
The issue is that I want one backup for all files belonging to the project.
I would have to extract the archive, bring the files together and then create a new archive before deleting the old, which can create unnecessary wear on the partitions and eat time.

New suggestion:
How about a `--extend` flag for `borg-create` that can take an existing archive and includes all its contents in the new archive.
Then I could use that and delete the original archive afterwards.
